concepts/pointers/about.md

```diff
@@ -0,0 +1,404 @@
+# About
+
+A pointer is a variable or constant whose value is the address of another variable or constant.
+Pointers are a powerful and potentially dangerous feature of C.
+It can be challenging to understand pointers.
+
+```c
+int i = 5;
+```
+
+When we assign a value to a variable we can then ask two questions.
+
+Who are you?
+
+_I am `i`_.
+
+What do you have?
+
+_I have the integer `5`_.
+
+There is also a third question we can ask.
+
+Where are you?
+
+_I am at `0x6dfed4`_.
+
+Wait, what?
+
+Variables and constants store their value in a memory location.
+That location in memory is known as the address of the variable or constant.
+
+## Address operator
+
+To get the address of a variable we use the address operator, which is `&`.
+
+
+```c
+int i = 5;
+int *i_address = &i;
+```
+
+Wait, what does that `*` mean before `i_address`? Doesn't that mean multiply?
+
+A type followed by a `*` followed by an identifier is the way to declare a pointer to a variable of that type.
+
+Okay, but why make a variable just to hold the address of another variable? 
+
+Why do we care what the address of a variable is in the first place?
+
+Well, we care where variables are when they are in an array.
+
+## Arrays and Pointers
+
+When we create an array like so
+
+```c
+int my_ints[] = {1, 2};
+```
+
+the value of `my_ints` is actually its address, which also happens to be the same as the address of its first element. The address of its second element will be the first element's address plus the size of an int on the system.
+
+To better see this, try running the following
+
+```c
+#include <stdio.h>
+
+int main() {
+    // my_int's value is its address
+    int my_ints[] = {1, 2};
+    
+    // since my_ints is already an address
+    // we don't need the address operator to make a pointer from it.
```

Comment:
Hold on, `*` is explained above, but `&` hasn't been explained yet...
Shouldn't that also be explained a little as part of the example on ln # 39?